torch/fx/experimental/symbolic_shapes.py

```diff
@@ -258,16 +300,21 @@ def magic_impl(self, other):
         other_expr = self.shape_env.replace(other_expr)
         out = func(expr, other_expr)
         out = sympy.expand(out)
-        if method in ["truediv"]:
+        if method in always_float_magic_methods:
```

Comment:
oh an extra assert in `PythonSymInt/FloatNodeImpl` constructors might be a good idea because of this - if someone updates this list, but forgets to update the corresponding output type of the methods on the `PythonSymIntNodeImpl` class, an assert would be nice.

Reply:
In the constructor, it is another issue as today we do allow `SymIntNode(PySymInt())` to contain anything: int, float or bool (same for the float version).
We could restrict this to int+bool and float+bool maybe but that might be breaking a bunch of stuff.